Mathlib/Analysis/Normed/Operator/ContinuousAlgEquiv.lean

```diff
@@ -79,3 +85,145 @@ public theorem ContinuousAlgEquiv.eq_continuousLinearEquivConjContinuousAlgEquiv
+  left_inv x := by
+    have := by simpa using congr($he x)
+    simp [smul_smul, hα2, this, hα, ← mul_assoc]
+  right_inv x := by
+    have := by simpa using congr($he' x)
+    simp [smul_smul, hα2, this, hα, ← mul_assoc]
+  continuous_toFun := (α' • e.toContinuousLinearMap).continuous
+  continuous_invFun := (α' • e.toContinuousLinearMap.adjoint).continuous
+
+@[simp] theorem coe_auxContinuousLinearEquiv :
+    (auxContinuousLinearEquiv e hα hα2 he he').toContinuousLinearMap =
+      α' • e.toContinuousLinearMap := rfl
+
+/-- Construct an isometry linear equivalence from a continuous linear equivalence
+and that its adjoint is a real-scalar multiple of its inverse. -/
+noncomputable abbrev auxIsometry (hαa : starRingEnd 𝕜 α' = α') :
+    V ≃ₗᵢ[𝕜] W where
+  __ := auxContinuousLinearEquiv e hα hα2 he he' |>.toLinearEquiv
+  norm_map' := by
+    rw [ContinuousLinearEquiv.coe_toLinearEquiv, ← ContinuousLinearEquiv.coe_coe,
+      norm_map_iff_adjoint_comp_self, coe_auxContinuousLinearEquiv,
+      MulActionSemiHomClass.map_smulₛₗ]
+    simp only [hαa, comp_smulₛₗ, RingHom.id_apply, smul_comp, smul_smul, hα2]
+    simp [he, smul_smul, hα, one_def]
+
+@[simp] theorem coe_auxIsometry (hαa : starRingEnd 𝕜 α' = α') :
+    (auxIsometry e hα hα2 he he' hαa).toContinuousLinearEquiv.toContinuousLinearMap =
+      α' • e.toContinuousLinearMap := rfl
+
+@[simp] theorem coe_symm_auxIsometry (hαa : starRingEnd 𝕜 α' = α') :
+    (auxIsometry e hα hα2 he he' hαa).toContinuousLinearEquiv.symm.toContinuousLinearMap =
+      α'⁻¹ • e.symm.toContinuousLinearMap := by
+  ext y
+  apply (auxIsometry e hα hα2 he he' hαa).toContinuousLinearEquiv.injective
+  simp [smul_smul, inv_mul_cancel₀ (a := α') (by grind)]
+
+end auxiliaryDefs
+
+open ComplexOrder
+
+/-- The ⋆-algebra equivalence version of
+`ContinuousAlgEquiv.eq_continuousLinearEquivConjContinuousAlgEquiv`.
+
+TODO: remove the hypothesis `Continuous f`, as star-algebra equivalences between endomorphisms are
+automatically continuous. -/
+public theorem StarAlgEquiv.eq_linearIsometryEquivConjStarAlgEquiv
+    (f : (V →L[𝕜] V) ≃⋆ₐ[𝕜] (W →L[𝕜] W)) (hf : Continuous f) :
+    ∃ U : V ≃ₗᵢ[𝕜] W, f = U.conjStarAlgEquiv := by
+  /- Assume nontriviality of `V`.
+    Then by `ContinuousAlgEquiv.eq_continuousLinearEquivConjContinuousAlgEquiv`,
+    we know there exists a continuous linear equivalence `y : V ≃L[𝕜] W` such that
+    `f = y.conjAlgEquiv`.
+    Our goal will be to construct an isometry from `y`. We do this by first showing
+    `adjoint y ∘ y` is in the center of the endormorphisms, and as the algebra of endomorphisms
+    are central, `adjoint y ∘ y` is a scalar multiple of the identity.
+    Let `α : 𝕜` be that scalar, i.e., `adjoint y ∘ y = α • id`. This scalar is clearly real.
+    Also, it's easy to see that `adjoint y ∘ y` is invertible, so `α ≠ 0`.
+    As `adjoint y ∘ y` is positive, we get `0 < α`.
+    We also get `y ∘ adjoint y = α • id`.
+    Finally, we construct our isometry `1/√(re α) • y`. -/
```

Comment:
This is nice, but it would be even nicer if it were interspersed throughout the code.